playbooks/roles/mongo_newrelic_monitor/tasks/main.yml

```diff
@@ -0,0 +1,45 @@
+---
+
+- name: Install newrelic mongo integration
+  apt:
+    pkg: "{{ item }}"
+    state: present
+    update_cache: yes
+  with_items:
+    - "nri-mongodb"
+
+- name: Create newrelic monitoring role
+  shell:
+    cmd: 'mongo --authenticationDatabase admin -u "{{ MONGO_ADMIN_USER }}" -p "{{ MONGO_ADMIN_PASSWORD }}" --host "{{ MONGO_REPL_SET }}/{{ EDXAPP_MONGO_HOSTS }}" --eval ''db.createRole({ role: "listCollections", privileges: [{ resource: {db:"",collection:""}, actions: ["listCollections"] }], roles: [] });'' admin'
+  register: create_newrelic_role_result
+  ignore_errors: true
+
+- name: Fail if create newrelic monitoring role actually failed
+  fail:
+    msg: create newrelic monitoring role failed
+  when: 'create_newrelic_role_result.rc != 0 and "already exists" not in create_newrelic_role_result.stdout'
```

Comment:
@itsjeyd Adding this allows the creating role command to be idempotent while still catching actual errors - if it already exists, it exits with a nonzero exit code, but we don't want to fail in that case, so we don't fail if "already exists" is present in stdout.

A neater method would be to check if the monitoring role exists before running the first mongo command, but in the interests of saving time, this was easier. If we have extra time, this is something I can look into if you like?

Reply:
@swalladge 

> Adding this allows the creating role command to be idempotent while still catching actual errors - if it already exists, it exits with a nonzero exit code, but we don't want to fail in that case, so we don't fail if "already exists" is present in stdout.

OK I see, thanks for the details :)

> A neater method would be to check if the monitoring role exists before running the first mongo command, but in the interests of saving time, this was easier. If we have extra time, this is something I can look into if you like?

We don't really have extra time so this is fine to leave as is for now. A good point in time to come back to this would be when upstreaming these changes :) It should be OK to do that via a follow-up ticket. Could you create one and put it into Sprint 211 for now? (Given that lots of people will be off in the second half of Sprint 210 I don't think it would fit into that sprint, realistically :))

Reply:
@itsjeyd created [SE-1983](https://tasks.opencraft.com/browse/SE-1983) for this. :)